src/content-tags-drawer/data/apiHooks.jsx

```diff
@@ -109,3 +114,13 @@ export const useContentDataResponse = (contentId) => {
 export const useIsContentDataLoaded = (contentId) => (
   useContentData(contentId).status === 'success'
 );
+
+/**
+ * Builds the mutation to update the tags applied to the content object
+ * @returns {import("./types.mjs").UseMutationResult}
+ */
+export const useContentTaxonomyTagsMutation = () => (
+  useMutation({
+    mutationFn: ({ contentId, taxonomyId, tags }) => updateContentTaxonomyTags(contentId, taxonomyId, tags),
```

Comment:
Here's how to get the right typing for `mutationFn`:

1. Remove the incomplete `@returns` type as mentioned above.
2. Add a type declaration to `mutationFn` of `@type {import("@tanstack/react-query").MutateFunction<any, any, {contentId: string, taxonomyId: number, tags: string[]}>}` as seen in the screenshot below.
3. Now all the types are accurate - TypeScript can infer the rest automatically. You can see in the screenshot that when you go to call `mutate()` it knows what arguments you need.

![Screenshot 2023-11-23 at 10 56 25 AM](https://github.com/openedx/frontend-app-course-authoring/assets/945577/39767c27-117d-4f2f-92e0-101686826236)

Note: I think `taxonomyId` should be a number, but you can do it as a string if you want.

Reply:
Updated as well (also changed it`taxonomyId` to a number), https://github.com/openedx/frontend-app-course-authoring/pull/704/commits/bdc8d3026150400e6d7b081e71e61180b83d5ff3